### PR TITLE
Feat(web-react): Introduce UNSTABLE_Truncate component #DS-1098

### PR DESCRIPTION
**Pull request by curdaj**

### CODE REVIEW

File: packages/web-react/src/components/UNSTABLE_Truncate/demo/index.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+import DocsSection from '../../../../docs/DocsSections';
+import TruncateDefault from './TruncateDefault';
+
+ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
+  <React.StrictMode>
+    <DocsSection title="Basic usage" stackAlignment="stretch">
```
Comment (crishpeen) [inline packages/web-react/src/components/UNSTABLE_Truncate/demo/index.tsx:8]: Title case, please.
```suggestion
    <DocsSection title="Basic Usage" stackAlignment="stretch">
```

File: packages/web-react/src/components/UNSTABLE_Truncate/stories/UNSTABLE_Truncate.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    docs: {
+      page: () => <Markdown>{ReadMe}</Markdown>,
+    },
+  },
+  argTypes: {
+    children: {
+      control: 'text',
+    },
+    lines: {
+      control: 'number',
```
Comment (crishpeen) [inline packages/web-react/src/components/UNSTABLE_Truncate/stories/UNSTABLE_Truncate.stories.tsx:20]: please add a line with the default value. See other stories on how to do it. (it is something like table default value or sth)

File: packages/web-react/src/components/UNSTABLE_Truncate/stories/UNSTABLE_Truncate.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    },
+  },
+  args: {
+    children:
```
Comment (crishpeen) [inline packages/web-react/src/components/UNSTABLE_Truncate/stories/UNSTABLE_Truncate.stories.tsx:24]: please add the `lines` here as well with the default value set

File: packages/web-react/src/components/UNSTABLE_Truncate/useTruncateStyleProps.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+export interface TruncateStyles<T extends ElementType> {
+  classProps: string;
+  props: SpiritTruncateProps<T>;
+  styleProps: TruncateCSSProperties;
+}
+
+export function useTruncateStyleProps<T extends ElementType>(props: SpiritTruncateProps<T>): TruncateStyles<T> {
+  const { lines, ...restProps } = props;
+
+  const TruncateMultilinesClass = 'text-truncate-multiline';
```
Comment (crishpeen) [inline packages/web-react/src/components/UNSTABLE_Truncate/useTruncateStyleProps.ts:17]: this class can be prefixed. Please add the prefix hook.

File: packages/web-react/src/components/UNSTABLE_Truncate/stories/UNSTABLE_Truncate.stories.tsx (hunk 3 above)
Comment (crishpeen) [reply packages/web-react/src/components/UNSTABLE_Truncate/stories/UNSTABLE_Truncate.stories.tsx:24]: if it isn't set here, it still appears in the table, but you have click once more to edit it.
Current state:
image
If the `lines` object is already set here:
image

File: packages/web-react/src/components/UNSTABLE_Truncate/demo/TruncateDefault.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            max="10"
+            value={lines}
+            onChange={(e) => setLines(Number(e.currentTarget.value))}
+            label="Number of truncated lines:"
+            name="linesNumber"
+            id="truncate-lines"
+          />
+        </fieldset>
+      </form>
+      <Text size="xsmall">Maximum number of lines for demo purposes is 10.</Text>
```
Comment (crishpeen) [inline packages/web-react/src/components/UNSTABLE_Truncate/demo/TruncateDefault.tsx:25]: why don't you use the helper text of the TextField? :)

**Review by crishpeen — CHANGES_REQUESTED** [review]
Please add and test the classname prefix. Then I have few more small adjustment suggestions :)

File: packages/web-react/src/components/UNSTABLE_Truncate/useTruncateStyleProps.ts (hunk 4 above)
Comment (curdaj) [reply packages/web-react/src/components/UNSTABLE_Truncate/useTruncateStyleProps.ts:17]: EDIT - solved. Thanks

File: packages/web-react/src/components/UNSTABLE_Truncate/stories/UNSTABLE_Truncate.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
+import { Markdown } from '@storybook/blocks';
+import type { Meta, StoryObj } from '@storybook/react';
+import React from 'react';
+import ReadMe from '../README.md';
+import { UNSTABLE_Truncate } from '..';
```
Comment (dlouhak) [inline packages/web-react/src/components/UNSTABLE_Truncate/stories/UNSTABLE_Truncate.stories.tsx:5]: ⚠️ I know that this not part of the production package, but it is more safe to use file import.

```suggestion
import UNSTABLE_Truncate from '../UNSTABLE_Truncate';
```

See https://github.com/lmc-eu/spirit-design-system/pull/1464.

**Review by dlouhak — COMMENTED** [review]
Looks great! 💪🏻

File: packages/web-react/src/components/UNSTABLE_Truncate/useTruncateStyleProps.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import { CSSProperties, ElementType } from 'react';
+import { SpiritTruncateProps } from '../../types/truncate';
+
+interface TruncateCSSProperties extends CSSProperties {
+  '--text-truncate-lines'?: number;
```
Comment (pavelklibani) [inline packages/web-react/src/components/UNSTABLE_Truncate/useTruncateStyleProps.ts:5]: maybe we could use some constant for `--text-truncate-lines`

File: packages/web-react/src/components/UNSTABLE_Truncate/README.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+> ⚠️ This component is UNSTABLE. It may significantly change at any point in the future.
+> Please use it with caution.
+
+Truncate is a component that truncate text based on defined criteria.
+
+```jsx
+import { UNSTABLE_Truncate } from '@lmc-eu/spirit-web-react';
+
+<UNSTABLE_Truncate>
+    <!-- Text go here -->
```
Comment (pavelklibani) [inline packages/web-react/src/components/UNSTABLE_Truncate/README.md:12]:
```suggestion
    {/* Text go here */}
```

I believe this comment type is only used in HTML.

File: packages/web-react/src/components/UNSTABLE_Truncate/README.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+</UNSTABLE_Truncate>
+```
+
+## Lines
+
+You can add the number of lines to which you want to truncate the text using `lines` props
+
+```jsx
+<UNSTABLE_Truncate lines={2}>
+    <!-- Text go here -->
```
Comment (pavelklibani) [inline packages/web-react/src/components/UNSTABLE_Truncate/README.md:22]:
```suggestion
    {/* Text go here */}
```

**Review by pavelklibani — APPROVED** [review]
👍

**Comment by pavelklibani** [comment]
Add the component to `/packages/web-react/scripts/entryPoints.js`

File: packages/web-react/src/components/UNSTABLE_Truncate/__tests__/UNSTABLE_Truncate.test.tsx [hunk 10]
```diff
@@ -0,0 +1,39 @@
+import '@testing-library/jest-dom';
+import { render, screen } from '@testing-library/react';
+import React from 'react';
+import { restPropsTest } from '../../../../tests/providerTests/restPropsTest';
+import { stylePropsTest } from '../../../../tests/providerTests/stylePropsTest';
+import UNSTABLE_Truncate from '../UNSTABLE_Truncate';
+
+describe('UNSTABLE_Truncate', () => {
+  stylePropsTest(UNSTABLE_Truncate);
```
[the remark this answers is not in the document]
Comment (crishpeen) [reply packages/web-react/src/components/UNSTABLE_Truncate/__tests__/UNSTABLE_Truncate.test.tsx:9]: classNamePrefixProviderTest(UNSTABLE_Truncate, 'text-truncate-multiline');

File: packages/web-react/src/components/UNSTABLE_Truncate/stories/UNSTABLE_Truncate.stories.tsx (hunk 2 above)
Comment (crishpeen) [reply packages/web-react/src/components/UNSTABLE_Truncate/stories/UNSTABLE_Truncate.stories.tsx:20]: nope, the default value, as in CSS is 3

File: packages/web-react/src/components/UNSTABLE_Truncate/__tests__/UNSTABLE_Truncate.test.tsx [hunk 11]
```diff
@@ -0,0 +1,41 @@
+import '@testing-library/jest-dom';
+import { render, screen } from '@testing-library/react';
+import React from 'react';
+import { restPropsTest } from '../../../../tests/providerTests/restPropsTest';
+import { stylePropsTest } from '../../../../tests/providerTests/stylePropsTest';
+import UNSTABLE_Truncate from '../UNSTABLE_Truncate';
```
Comment (dlouhak) [inline packages/web-react/src/components/UNSTABLE_Truncate/__tests__/UNSTABLE_Truncate.test.tsx:7]: I would move this import down to group imports by relative paths, but I'm not sure if you have this non-automated formatting rule.

**Review by crishpeen — APPROVED** [review]

**Review by dlouhak — APPROVED** [review]

**Pull request merged**